### ISSUE TITLE
How can I set project properties?

### ISSUE THREAD

**Issue by jay0129** [gradle/gradle-profiler#446]
In your example, I can see how to set system-properties.
However, there is no project-properties example.

**Comment by gabrielfeo** [comment]
You can set project properties via system properties by prefixing them with `org.gradle.project.` ([docs](https://docs.gradle.org/current/userguide/build_environment.html#sec:project_properties), example 14)

```properties
org.gradle.project.foo=bar
```

```conf
my-scenario {
  system-properties {
    "org.gradle.project.foo" = "bar"
  }
}
```

**Comment by notsatyarth** [comment]
I've been unsuccessful in setting this up. Passing the properties in this manner is not received by the project. Has anyone got this working successfully?

**Comment by gabrielfeo** [comment]
True. Not even this works:

```conf
my-scenario {
  gradle-args = [
    "-Pfoo=bar"
  ]
}
```

Maybe maintainers can shed a light.

**Comment by asodja** [comment]
Tested with `gradle-args` and works ok.

See: https://github.com/asodja/gradle-profiler-properties-test
Scenario file: https://github.com/asodja/gradle-profiler-properties-test/blob/main/performance.scenario
Build file: https://github.com/asodja/gradle-profiler-properties-test/blob/main/build.gradle.kts

When I run:
```
gradle-profiler --benchmark  --scenario-file ./performance.scenario
cat profile-out/profile.log | grep -E "myGradleProperty|mySystemProperty" 
```

I get always:
```
Value of myGradleProperty=hello
Value of mySystemProperty=world
```

**Comment by gabrielfeo** [comment]
OP asked about **project** properties (which I tested with `-P` and `val foo by project.properties`), but yes, migrating to other types is a good suggestion :)

I'm curious as to why project properties don't work though, even with `--cli`.

**Comment by asodja** [comment]
I see. Note, that in example I am also using `-PmyGradleProperty=hello`. 
I added examples also for other cases and it seems they work :
https://github.com/asodja/gradle-profiler-properties-test/blob/917cc06f931d0049ede054085ceb3faf6675c08f/build.gradle.kts#L4-L9
```
// This will blow up if no default value is set
val myGradleProperty: String by project.properties
println("Value of myGradleProperty: String by project.properties=${myGradleProperty}")
// This will blow up if no default value is set
println("Value of project.property(\"myGradleProperty\")=${project.property("myGradleProperty").toString()}")
println("Value of properties[\"myGradleProperty\"]=${properties["myGradleProperty"]}")
```

Note that:
```
val myGradleProperty: String by project.properties
or
val myGradleProperty by project
```
and
```
project.property("myGradleProperty").toString()
```
will not work if you don't have some default value set (see [gradle.properties](https://github.com/asodja/gradle-profiler-properties-test/blob/main/gradle.properties)), since gradle-profiler first inspects your build which will evaluate also projects. And probably that causes some problems.

---

In the end I think this is more of documentation issue, where we should tell users that if you use some properties that needs to be always set, you should set some default values for in gradle.properties file.

**Comment by gabrielfeo** [comment]
Good catch, even system properties blow up if there's no default. I'll open a PR suggesting changes to the README.

**Comment by mrk-han** [comment]
Note: If your project was using any project properties from the user's global gradle properties file `~/.gradle/gradle.properties` you may need to copy those over to the new gradle-user-home location (default root of project) so `~/project-root/gradle-user-home/gradle.properties`
